### PR TITLE
Add project epics to issues and pull requests

### PR DESCRIPTION
**Pull request by boazreicher**
This PR formats the way that issues and pull requests are presented.  Instead of using a simple markdown list, they are formatted into a markdown table, where the second column shows the epics (and corresponding projects) for each item (for pull requests, we use the epics for any linked issues that the PR will close)

### CODE REVIEW

File: src/makeGroupsIntoMarkdown.ts [hunk 1]
```diff
@@ -25,23 +25,47 @@ const makeGroupsIntoMarkdown = (
   for (const groups of groupGroups) {
     for (const primaryVal of Object.values(groups.primary)) {
       primaryArr.push(`### ${primaryVal.groupTitle}\n`);
+      if (primaryVal.itemType === "PR" || primaryVal.itemType === "Issue") {
```
Comment (inkblotty) [inline src/makeGroupsIntoMarkdown.ts:28]: Can we abstract the table formatting into a helper function to preserve readability here?

File: src/queries.ts [hunk 2]
```diff
@@ -15,6 +15,26 @@ query getUserWork($username:String!, $owner:String!, $repo:String!, $sinceIso: D
...
+          closingIssuesReferences(first: 10) {
+            edges {
+              node {
+                projectItems(first: 10) {
+                  edges {
+                    node {
+                      project {
+                        title
+                      }
+                      fieldValueByName(name: "Epic") {
```
Comment (inkblotty) [inline src/queries.ts:27]: Can we make this a configurable field that the user can set? Not every team may use Epics to group issues.

**Review by inkblotty — COMMENTED** [review]
This is awesome. I think I'd prefer a little more configurability in case someone wants to group by a different issue relationship than "Epic", and a note on readability. I don't the resources introduced will be too much, but we'll have to keep an eye on it.

**Comment by boazreicher** [comment]
@inkblotty addressed your comments in https://github.com/inkblotty/my-work-action/pull/21/commits/5876c274618d0f8e378f354359f4b0c9065514fd

File: README.md [hunk 3]
```diff
@@ -40,3 +40,6 @@ Number of days to query, with today as the end date. Defaults to 7 days.
 
 - `usernames`
 Your user handle, or your list of teammates, comma-separated.
+
+- `project_field`
+A project field to display for issues and pull requests.
```
Comment (inkblotty) [inline README.md:45]:
```suggestion
A project field to display for issues and pull requests. For example, "Epic" for relationships with "Epic" issues.
```

Is this right? It reminds me of the literal project board when you say "Project", but maybe I misunderstand how this query works.

Comment (boazreicher) [reply README.md:45]: By `project field` I mean a (single select) field defined in a Project, like these (`Status` and `Type`):
image

What the query does, is basically the following:

1. For every issue and pull request:
2. For every project the issue/PR belongs to:
3. Extract the value of the `${project_field}` single select field (if such a field exists in the project)

Comment (inkblotty) [reply README.md:45]: Ok that makes total sense. Let me try again.

```suggestion
A project field to display for issues and pull requests. This is usually a column (on Table views) from a GitHub Project, such as Status or Iteration.
```

File: src/queries.ts [hunk 4]
```diff
@@ -107,7 +161,9 @@ fragment repo on Repository {
     }
   }
 `;
-export const getAllWorkForRepository = async (requestOwner: string, repoName: string, username: string, sinceIso: string): Promise<{ [key: string]: QueryGroup }> => {
+export const getAllWorkForRepository = async (requestOwner: string, repoName: string, username: string, sinceIso: string, projectField?: string): Promise<{ [key: string]: QueryGroup }> => {
+    const projectFieldVariables = projectField ? { addProjectFields: true, projectField } : {};
+    console.log("[BR] projectFieldVariables", projectFieldVariables)
```
Comment (inkblotty) [inline src/queries.ts:166]:
```suggestion
```

**Review by inkblotty — APPROVED** [review]
Approved with suggestion :) Feel free to merge and test on your own workflow when you have a chance.
